test/index.js

```diff
@@ -15,167 +15,138 @@ const internals = {};
 
-    it('fails if keepAlive is configured but not ttl', (done) => {
+    it('fails if keepAlive is configured but not ttl', async () => {
 
         const server = new Hapi.Server();
         server.connection();
-        server.register(require('../'), (err) => {
 
-            expect(err).to.not.exist();
+        await server.register(require('../'));
 
-            expect(() => {
+        expect(() => {
 
-                server.auth.strategy('default', 'cookie', true, {
-                    password: 'password-should-be-32-characters',
-                    keepAlive: true
-                });
-            }).to.throw(Error);
-
-            done();
-        });
+            server.auth.strategy('default', 'cookie', true, {
+                password: 'password-should-be-32-characters',
+                keepAlive: true
+            });
+        }).to.throw(Error);
     });
 
-    it('authenticates a request', (done) => {
+    it('authenticates a request', async () => {
 
         const server = new Hapi.Server();
         server.connection();
-        server.register(require('../'), (err) => {
 
-            expect(err).to.not.exist();
+        await server.register(require('../'));
 
-            server.auth.strategy('default', 'cookie', true, {
-                password: 'password-should-be-32-characters',
-                ttl: 60 * 1000,
-                domain: 'example.com',
-                cookie: 'special',
-                clearInvalid: true,
-                validateFunc: function (request, session, callback) {
+        server.auth.strategy('default', 'cookie', true, {
+            password: 'password-should-be-32-characters',
+            ttl: 60 * 1000,
+            domain: 'example.com',
+            cookie: 'special',
+            clearInvalid: true,
+            validateFunc: function (request, session, callback) {
 
-                    const override = Hoek.clone(session);
-                    override.something = 'new';
+                const override = Hoek.clone(session);
+                override.something = 'new';
 
-                    return callback(null, session.user === 'valid', override);
-                }
-            });
+                return callback(null, session.user === 'valid', override);
+            }
+        });
 
-            server.route({
-                method: 'GET', path: '/login/{user}',
-                config: {
-                    auth: { mode: 'try' },
-                    handler: function (request, reply) {
+        server.route({
+            method: 'GET', path: '/login/{user}',
+            config: {
+                auth: { mode: 'try' },
+                handler: function (request, reply) {
 
-                        request.cookieAuth.set({ user: request.params.user });
-                        return reply(request.params.user);
-                    }
+                    request.cookieAuth.set({ user: request.params.user });
+                    return reply(request.params.user);
```

Comment:
`reply` is not a v17 thing anymore, I'm surprised it would work.

Reply:
He said it's not intended to be released